core/src/main/java/org/wildfly/extras/creaper/core/online/HttpModelControllerClient.java

```diff
@@ -0,0 +1,145 @@
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.auth.DigestScheme;
+import org.apache.http.impl.client.BasicAuthCache;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.Operation;
+import org.jboss.as.controller.client.OperationMessageHandler;
+import org.jboss.as.controller.client.OperationResponse;
+import org.jboss.dmr.ModelNode;
+import org.jboss.threads.AsyncFuture;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+/**
+ * Controller provides execution of {@link ModelNode} or {@link Operation} over HTTP.
+ * Asynchronous execution is not provided. Controller does not inform about any progress, therefore, there is no point
+ * in using {@link OperationMessageHandler}.
+ * If an error occurs (server does not respond 401 or header does not contain WWW-Authenticate field after
+ * first request)
+ * {@link IllegalStateException} is thrown.
+ */
+public class HttpModelControllerClient implements ModelControllerClient {
+    private HttpHost target;
+    private CloseableHttpClient httpClient;
+    private HttpClientContext localContext;
+    private final String url;
+
+    public HttpModelControllerClient(String host, int port, String username, String password) {
+        target = new HttpHost(host, port, "http");
+        CredentialsProvider credsProvider = new BasicCredentialsProvider();
+        credsProvider.setCredentials(
+                new AuthScope(target.getHostName(), target.getPort()),
+                new UsernamePasswordCredentials(username, password));
+
+        httpClient = HttpClients.custom()
+                .setDefaultCredentialsProvider(credsProvider)
+                .build();
+        url = String.format("http://%s:%s/management", host, port);
+    }
+
+    private void initialize() {
+        localContext = HttpClientContext.create();
+    }
+
+    @Override
+    public ModelNode execute(ModelNode modelNode) throws IOException {
+        initialize();
+        try {
+            AuthCache authCache = new BasicAuthCache();
+            authCache.put(target, getDigestScheme());
+            localContext.setAuthCache(authCache);
+            CloseableHttpResponse response = httpClient.execute(target, buildRequest(modelNode), localContext);
+            ModelNode m = ModelNode.fromJSONStream(response.getEntity().getContent());
+            response.close();
+            return m;
+
+        } catch (MalformedChallengeException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    @Override
+    public ModelNode execute(Operation operation) throws IOException {
+        return execute(operation.getOperation());
```

Comment:
`Operation` can have `InputStream`s attached. There's a way to translate that to HTTP, we should figure it out. Or at least throw an exception if there are attachments.